service/tablet_allocator.cc

```diff
@@ -1149,6 +1150,32 @@ class load_balancer {
     }
 };
 
+future<tablet_replica_calculation_result>
+calculate_tablet_replicas_for_new_rf(const tablet_aware_replication_strategy* rep, schema_ptr s, token_metadata_ptr tm,
+    std::map<sstring, sstring> ks_options) {
+
+    auto table_id = s->id();
+    tablet_map old_tablets = tm->tablets().get_tablet_map(table_id);
+
+    if (dynamic_cast<const locator::network_topology_strategy*>(rep) == nullptr) {
+        co_return tablet_replica_calculation_result{ old_tablets, tablet_replica_calculation_status::unknown_topology_strategy };
+    }
+
+    locator::replication_strategy_params params{ks_options, old_tablets.tablet_count()};
+
+    auto new_rep = abstract_replication_strategy::create_replication_strategy(
+                "NetworkTopologyStrategy", params);
+
+    auto tablet_aware = new_rep->maybe_as_tablet_aware();
+
+    if (!tablet_aware) {
```

Comment:
again, this should be an internal error.
We're over engineering the interface with this `tablet_replica_calculation_result`.
The function returns a future that already can represent both the success/failure status and an error when needed

Reply:
Cc @KrzaQ

Reply:
I agree, the whole function has been removed.